Extension/src/LanguageServer/settings.ts

```diff
@@ -996,28 +860,71 @@ export class OtherSettings {
+    }
+
+    private getAsBoolean(sectionName: string, settingName: string, resource: any, defaultBoolean: boolean): boolean {
+        const fullConfiguration = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = fullConfiguration.get<boolean>(settingName);
+        if (isBoolean(value)) {
+            return value;
+        }
+        const section = fullConfiguration.inspect<boolean>(settingName);
+        return section ?.defaultValue ?? defaultBoolean;
     }
-    public get filesExclude(): vscode.WorkspaceConfiguration | undefined { return vscode.workspace.getConfiguration("files", this.resource).get("exclude"); }
-    public get filesAutoSaveAfterDelay(): boolean { return vscode.workspace.getConfiguration("files").get("autoSave") === "afterDelay"; }
-    public get editorInlayHintsEnabled(): boolean { return vscode.workspace.getConfiguration("editor.inlayHints").get<string>("enabled") !== "off"; }
-    public get editorParameterHintsEnabled(): boolean | undefined { return vscode.workspace.getConfiguration("editor.parameterHints").get<boolean>("enabled"); }
-    public get searchExclude(): vscode.WorkspaceConfiguration | undefined { return vscode.workspace.getConfiguration("search", this.resource).get("exclude"); }
-    public get workbenchSettingsEditor(): string | undefined { return vscode.workspace.getConfiguration("workbench.settings").get<string>("editor"); }
 
-    public get colorTheme(): string | undefined { return vscode.workspace.getConfiguration("workbench").get<string>("colorTheme"); }
+    private getAsNumber(sectionName: string, settingName: string, resource: any, defaultNumber: number): number {
+        const fullConfiguration = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = fullConfiguration.get<number>(settingName);
+        if (isNumber(value)) {
+            return value;
+        }
+        const section = fullConfiguration.inspect<number>(settingName);
+        return section ?.defaultValue ?? defaultNumber;
+    }
 
-    public getCustomColorToken(colorTokenName: string): string | undefined { return vscode.workspace.getConfiguration("editor.tokenColorCustomizations").get<string>(colorTokenName); }
-    public getCustomThemeSpecificColorToken(themeName: string, colorTokenName: string): string | undefined { return vscode.workspace.getConfiguration(`editor.tokenColorCustomizations.[${themeName}]`, this.resource).get<string>(colorTokenName); }
+    private getAsAssociations(sectionName: string, settingName: string, resource?: any): Associations {
+        const fullConfiguration = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = fullConfiguration.get<any>(settingName);
+        if (isValidMapping(value, 'string', 'string')) {
+            return value as Associations;
+        }
+        const section = fullConfiguration.inspect<any>(settingName);
+        return section ?.defaultValue as Associations;
```

Comment:
I don't understand this -- what if the default value is `null`?

Reply:
I missed that, if the default value is null, the function should be able to return null. I will refactor appropriately.

Reply:
The default associations are not null.  They shouldn't ever be null.  But we can pass in a default here like the other functions that matches the current default.